module/zcommon/zfs_fletcher.c

```diff
@@ -245,7 +214,107 @@ fletcher_4_incremental_byteswap(const void *buf, uint64_t size,
 	ZIO_SET_CHECKSUM(zcp, a, b, c, d);
 }
 
+static const struct fletcher_4_calls fletcher_4_generic_calls = {
+	.init = fletcher_4_generic_init,
+	.compute = fletcher_4_generic,
+	.compute_byteswap = fletcher_4_generic_byteswap,
+	.name = "generic"
+};
+
+static const struct fletcher_4_calls *chosen = &fletcher_4_generic_calls;
```

Comment:
Sure. In that case, it would be enough for ```zfs_fletcher_4_math_impl``` to be selected from one of "fastest", "original", and "avx2".